File: SCT2/trunk/test-plugins/org.yakindu.sct.generator.c.test/gtests/ConstantsTests/ConstantsTests.cc
/**
* Copyright (c) 2015 committers of YAKINDU and others.
* All rights reserved. This program and the accompanying materials
* are made available under the terms of the Eclipse Public License v1.0
* which accompanies this distribution, and is available at
* http://www.eclipse.org/legal/epl-v10.html
*
* Contributors:
*     committers of YAKINDU - initial API and implementation
*/
#include <string>
#include "gtest/gtest.h"
#include "Constants.h"


Constants handle;

TEST(StatemachineTest, constantDefinition) {
	constants_init(&handle);
	constants_enter(&handle);
	EXPECT_TRUE(constants_isActive(&handle, Constants_main_region_A));
	EXPECT_TRUE(constantsIface_get_x(&handle) == 10l);
	EXPECT_TRUE(constantsIface_get_y(&handle) == 20l);
	EXPECT_TRUE(strcmp(constantsIfaceNamed_get_y(&handle) , "Hello World") == 0);
	constantsIface_raise_e(&handle);
	constants_runCycle(&handle);
	EXPECT_TRUE(constantsIface_get_result(&handle) == 20l);
	constantsIface_raise_e(&handle);
	constants_runCycle(&handle);
	EXPECT_TRUE(constants_isActive(&handle, Constants_main_region_C));
	EXPECT_TRUE(constantsIface_get_result(&handle) == 100l);
	constantsIface_raise_e2(&handle, constantsIface_get_x(&handle) );
	constants_runCycle(&handle);
	EXPECT_TRUE(constantsIface_get_result(&handle) == 1000l);
	EXPECT_TRUE(constants_isActive(&handle, Constants_main_region_A));
}

		

File: test-plugins/org.yakindu.sct.generator.java.test/test-gen/org/yakindu/sct/generator/java/test/ConstantsTests.java
/**
* Copyright (c) 2017 committers of YAKINDU and others.
* All rights reserved. This program and the accompanying materials
* are made available under the terms of the Eclipse Public License v1.0
* which accompanies this distribution, and is available at
* http://www.eclipse.org/legal/epl-v10.html
*
* Contributors:
*     committers of YAKINDU - initial API and implementation
*/

package org.yakindu.sct.generator.java.test;

import org.junit.*;
import static org.junit.Assert.*;
import org.yakindu.scr.constants.ConstantsStatemachine;
import org.yakindu.scr.constants.ConstantsStatemachine.State;	

/**
 * Unit TestCase for Constants
 */
@SuppressWarnings("all")
public class ConstantsTests {
	
	private ConstantsStatemachine statemachine;	
	
	
	
	@Before
	public void constantsTests_setUp() {
		statemachine = new ConstantsStatemachine();
		statemachine.init();
	}

	@After
	public void constantsTests_tearDown() {
		statemachine = null;
	}
	
	@Test
	public void constantDefinition() {
		statemachine.enter();
		assertTrue(statemachine.isStateActive(State.main_region_A));
		assertTrue(statemachine.getX() == 10l);
		assertTrue(statemachine.getY() == 20l);
		assertTrue(statemachine.getSCINamed().getY().equals("Hello World"));
		statemachine.raiseE();
		statemachine.runCycle();
		assertTrue(statemachine.getResult() == 20l);
		statemachine.raiseE();
		statemachine.runCycle();
		assertTrue(statemachine.isStateActive(State.main_region_C));
		assertTrue(statemachine.getResult() == 100l);
		statemachine.raiseE2(statemachine.getX());
		statemachine.runCycle();
		assertTrue(statemachine.getResult() == 1000l);
		assertTrue(statemachine.isStateActive(State.main_region_A));
	}
}
